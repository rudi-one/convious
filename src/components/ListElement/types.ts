import {AppCollection} from 'store/apps/types';
import {MusicCollection} from 'store/music/types';

interface ElementProps {
  imageUrl: string;
  item: AppCollection | MusicCollection;
  showArtist?: boolean;
  navigate: () => void;
}

export default ElementProps;
