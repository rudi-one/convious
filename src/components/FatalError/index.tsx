import React from 'react';
import {Text, View} from 'react-native';
import stylesCommon from 'src/common/style';

const FatalError: React.FC = () => {
  return (
    <View style={stylesCommon.container}>
      <Text>Critical error</Text>
      <Text>'Please reload the application'</Text>
    </View>
  );
};

export default FatalError;
