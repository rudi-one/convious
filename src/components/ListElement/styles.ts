import {StyleSheet} from 'react-native';
import {Colors} from 'react-native-paper';
import {horizontalScale, moderateScale, verticalScale} from 'src/utils/helpers';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: '100%',
    flexDirection: 'row',
    justifyContent: 'flex-start',
    padding: moderateScale(5),
    backgroundColor: Colors.white,
    borderRadius: 10,
    marginBottom: verticalScale(10),
  },
  image: {
    width: horizontalScale(80),
    height: verticalScale(80),
    borderRadius: 10,
  },
  imageWrapper: {
    padding: moderateScale(5),
    borderRadius: 10,
    borderColor: Colors.grey400,
  },
  information: {
    flex: 1,
    justifyContent: 'center',
    paddingLeft: verticalScale(10),
  },
  genre: {
    fontSize: 15,
    color: Colors.cyan800,
  },
  title: {
    fontSize: 20,
    paddingVertical: verticalScale(5),
  },
});

export default styles;
