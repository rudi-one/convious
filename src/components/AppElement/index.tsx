import {useNavigation} from '@react-navigation/native';
import {NavigationProp, RouteNames} from 'navigation/types';
import React, {memo} from 'react';
import {Image, Text, TouchableOpacity, View} from 'react-native';
import AppProps from 'src/components/AppElement/types';
import styles from './styles';

const AppElementComponent = ({app}: AppProps) => {
  const navigation = useNavigation<NavigationProp>();

  return (
    <TouchableOpacity
      style={styles.container}
      onPress={() => {
        navigation.navigate(RouteNames.app, {app: app});
      }}>
      <View style={styles.imageWrapper}>
        <Image
          source={{
            uri: app.artworkUrl512,
          }}
          style={styles.image}
        />
      </View>
      <View style={styles.information}>
        <Text style={styles.genre}>{app.primaryGenreName}</Text>
        <Text style={styles.title}>{app.trackName}</Text>
      </View>
    </TouchableOpacity>
  );
};

export const AppElement = memo(AppElementComponent);
