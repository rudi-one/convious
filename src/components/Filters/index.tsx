import * as React from 'react';
import {Dispatch, memo, SetStateAction, useEffect, useRef} from 'react';
import {Pressable, ScrollView, View} from 'react-native';
import {Button, Colors, Dialog, IconButton, Text} from 'react-native-paper';
import {Filter} from 'src/components/Filters/types';
import styles from './styles';

interface FiltersProps {
  visible: boolean;
  setVisible: Dispatch<SetStateAction<boolean>>;
  filters: Filter[];
  filter: (item: Filter) => void;
  clear: (checked: boolean) => void;
}

const FiltersComponent = ({
  visible,
  setVisible,
  filters,
  filter,
  clear,
}: FiltersProps) => {
  const hideDialog = () => setVisible(false);
  const ref = useRef<ScrollView | null>(null);

  useEffect(() => {
    if (visible) {
      ref.current?.flashScrollIndicators();
    }
  }, [visible]);

  return (
    <Dialog visible={visible} onDismiss={hideDialog} style={styles.dialog}>
      <Dialog.Title>Filters (click to filter)</Dialog.Title>
      <Dialog.ScrollArea>
        <ScrollView
          contentContainerStyle={{paddingHorizontal: 14}}
          showsVerticalScrollIndicator={true}
          ref={ref}>
          {filters?.map(item => (
            <Pressable
              onPress={() => filter({name: item.name, checked: !item.checked})}
              key={item.name}>
              <View style={styles.checkboxWrapper}>
                <Text style={styles.checkboxLabel}>{item.name}</Text>
                <IconButton
                  icon={
                    item.checked ? 'checkbox-marked' : 'checkbox-blank-outline'
                  }
                  size={24}
                  color={Colors.greenA700}
                  style={styles.checkbox}
                />
              </View>
            </Pressable>
          ))}
        </ScrollView>
      </Dialog.ScrollArea>
      <Dialog.Actions>
        <Button onPress={() => clear(false)}>Clear</Button>
        <Button onPress={() => clear(true)}>Select all</Button>
        <Button onPress={hideDialog}>Done</Button>
      </Dialog.Actions>
    </Dialog>
  );
};

export const Filters = memo(FiltersComponent);
