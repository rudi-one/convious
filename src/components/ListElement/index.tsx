import React, {memo} from 'react';
import {Image, Text, TouchableOpacity, View} from 'react-native';
import ElementProps from 'src/components/ListElement/types';
import styles from './styles';

const ListElementComponent = ({
  item,
  imageUrl,
  showArtist,
  navigate,
}: ElementProps) => {
  return (
    <TouchableOpacity style={styles.container} onPress={navigate}>
      <View style={styles.imageWrapper}>
        <Image
          source={{
            uri: imageUrl,
          }}
          style={styles.image}
        />
      </View>
      <View style={styles.information}>
        <Text style={styles.genre}>{item.primaryGenreName}</Text>
        <Text style={styles.title}>
          {showArtist
            ? `${item.artistName}: ${item.trackName}`
            : item.trackName}
        </Text>
      </View>
    </TouchableOpacity>
  );
};

export const ListElement = memo(ListElementComponent);
