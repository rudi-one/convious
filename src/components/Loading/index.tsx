import React from 'react';
import {ActivityIndicator, Text, View} from 'react-native';
import {Colors} from 'react-native-paper';
import stylesCommon from 'src/common/style';
import {LoadingProps} from 'src/components/Loading/types';
import styles from './styles';

const Loading = ({mode}: LoadingProps) => {
  return (
    <View
      style={[
        stylesCommon.container,
        mode === 'absolute' ? styles.absolute : null,
      ]}>
      <ActivityIndicator
        animating={true}
        color={Colors.red800}
        size={'large'}
      />
      <Text style={styles.text}>loading...</Text>
    </View>
  );
};
export default Loading;
