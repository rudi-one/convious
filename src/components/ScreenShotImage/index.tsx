import React, {memo} from 'react';
import {Image, View} from 'react-native';
import styles from './styles';
import {ScreenShotProp} from './types';

const ScreenShotImageComponent = ({item}: {item: ScreenShotProp}) => {
  return (
    <View style={styles.screenshotContainer}>
      <Image
        source={{
          uri: item.url,
        }}
        style={styles.screenshot}
      />
    </View>
  );
};

export const ScreenShotImage = memo(ScreenShotImageComponent);
