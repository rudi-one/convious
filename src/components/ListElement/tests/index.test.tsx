import {render} from '@testing-library/react-native';
import React from 'react';
import {ListElement} from 'src/components/ListElement';
import {mockAppCollection1} from 'src/mocks/mockApps';

describe('<ListElement /> tests', () => {
  it('should match ListElement component snapshot', () => {
    const component = (
      <ListElement
        item={mockAppCollection1}
        imageUrl={''}
        navigate={() => {}}
      />
    );

    expect(render(component)).toMatchSnapshot();
  });
});
