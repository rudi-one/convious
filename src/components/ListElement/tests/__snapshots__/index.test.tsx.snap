// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListElement /> tests should match ListElement component snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 10,
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "marginBottom": 19.5,
      "opacity": 1,
      "padding": 13.5,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "borderColor": "#bdbdbd",
        "borderRadius": 10,
        "padding": 13.5,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "",
        }
      }
      style={
        Object {
          "borderRadius": 10,
          "height": 157,
          "width": 171.5,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "paddingLeft": 19.5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#00838f",
          "fontSize": 15,
        }
      }
    >
      Travel
    </Text>
    <Text
      style={
        Object {
          "fontSize": 20,
          "paddingVertical": 10,
        }
      }
    >
      TripIt: Travel Planner
    </Text>
  </View>
</View>
`;
